src/swap/coinzark.js

```diff
@@ -0,0 +1,272 @@
+}
+
+async function getAddress (
+  wallet: EdgeCurrencyWallet,
+  currencyCode: string
+): Promise<string> {
+  const addressInfo = await wallet.getReceiveAddress({ currencyCode })
+  return addressInfo.legacyAddress && !dontUseLegacy[currencyCode]
+    ? addressInfo.legacyAddress
+    : addressInfo.publicAddress
+}
+
+export function makeCoinZarkPlugin (
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { io, initOptions } = opts
+  const fetchJson = getFetchJson(opts)
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote (
+      request: EdgeSwapRequest,
+      userSettings: Object | void
+    ): Promise<EdgeSwapPluginQuote> {
+      const {
+        fromCurrencyCode,
+        fromWallet,
+        nativeAmount,
+        quoteFor,
+        toCurrencyCode,
+        toWallet
+      } = request
+      if (toCurrencyCode === fromCurrencyCode) {
+        throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+      }
+
+      if (quoteFor !== 'from') {
+        // CoinZark does not support reverse quotes
+        throw new SwapCurrencyError(swapInfo, fromCurrencyCode, toCurrencyCode)
+      }
+
+      // Grab addresses:
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(fromWallet, fromCurrencyCode),
+        getAddress(toWallet, toCurrencyCode)
+      ])
+
+      // Convert amount to CoinZark supported format
+      const quoteAmount = await fromWallet.nativeToDenomination(
+        nativeAmount,
+        fromCurrencyCode
+      )
+
+      // Convenience function to get JSON from the API
+      async function get (path: string) {
+        const api = `${uri}${path}`
+        const reply = await fetchJson(api)
+        return reply.json
+      }
+
+      // Convenience function to post form values and get returned JSON from the API
+      async function post (url, values: any) {
+        const opts = {
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            Accept: 'application/json'
+          },
+          method: 'POST',
+          body: ''
+        }
+        const formData = new URLSearchParams()
+        for (const prop in values) {
+          if (!values.hasOwnProperty(prop)) continue
+          formData.append(prop, values[prop])
+        }
+        opts.body = formData.toString()
+        const reply = await fetchJson(`${uri}${url}`, opts)
+        const out = reply.json
+        return out
+      }
+
+      // Fetch the supported currencies
+      const currencies = await get('swap/currencies')
+      let fromCorrect = false
+      let toCorrect = false
+
+      // Loop through the currencies and find the requested ones.
+      // CoinZark will return canDeposit / canReceive as status of the
+      // coins. The coin we want to exchange from should have canDeposit enabled
+      // and the coin we want to exchange to should have canReceive enabled.
+      if (!(currencies === null || currencies.result === null)) {
```

Comment:
We prefer double-equals with null (` == null`). This is the one exception to the `===` everywhere rule, since the `==` catches both null and undefined, and basically means "doesn't exist".

Reply:
Actually, reading it again, `(currencies != null && currencies.result != null)` might be even more obvious.

Reply:
Agreed, i've changed this.